shared-plugins/vip-go-elasticsearch/vip-go-elasticsearch.php

```diff
@@ -126,7 +126,7 @@ public function filter__the_posts( $posts, $query ) {
 		if ( ! $query->is_main_query() || ! $query->is_search() )
 			return $posts;
 
-		if ( ! is_array( $this->search_result ) )
+		if ( is_wp_error( $this->search_result ) || ! is_array( $this->search_result ) || empty( $this->search_result['results'] ) || empty( $this->search_result['results']['hits'] ) )
```

Comment:
Yes - `$this->search_result['results']['hits']` can be an empty array under normal circumstances, which would trigger the `empty()` check and incorrectly return the pre-filtered `$posts` instead of an empty array.

In practice the pre-filtered `$posts` is probably correct as-is due to how all the filters work in this plugin...but if we were to switch to `posts_pre_query` like we do in Jetpack Search, this would break and cause the query to fall back to a regular db query, as `$posts` will be `null` in that situation.

IMO it's better to handle an empty array of hits separately (since it's normal, expected behavior), explicitly returning an empty array like this:

https://github.com/Automattic/jetpack/blob/ee633d91d353eef2da953fd21791913121cfc45d/modules/search/class.jetpack-search.php#L187-L189